tools/src/wyvern/target/corewyvernIL/decltype/EffectDeclType.java

```diff
@@ -0,0 +1,113 @@
+package wyvern.target.corewyvernIL.decltype;
+
+import wyvern.target.corewyvernIL.IASTNode;
+
+import java.io.IOException;
+import java.util.HashSet;
+
+import wyvern.target.corewyvernIL.astvisitor.ASTVisitor;
+import wyvern.target.corewyvernIL.expression.Effect;
+import wyvern.target.corewyvernIL.support.TypeContext;
+import wyvern.target.corewyvernIL.support.View;
+import wyvern.target.corewyvernIL.type.ValueType;
+import wyvern.tools.errors.FileLocation;
+
+
+public class EffectDeclType extends DeclType implements IASTNode {
+	HashSet<Effect> effectSet;
```

Comment:
Use Set instead of HashSet in variable types.  It's better to reference the particular implementation of Set only when you create the HashSet.  Also make private.